### PR TITLE
Feature/error reporter

### PR DESCRIPTION
**Pull request by daiLlew**
Integration of import reporter client.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -44,18 +52,93 @@ func main() {
...
+		eventLoopCancel()
+		<-eventLoopDone
+		log.Debug("closing http server", nil)
+		if err := api.Close(ctx); err != nil {
+			log.ErrorC("failed to gracefully close http server", err, nil)
+		} else {
+			log.Debug("gracefully closed http server", nil)
+		}
+		log.Debug("closing kafka producer", nil)
+		service.Producer.Close(ctx)
```
Comment (nshumoogum) [inline main.go]: This can be renamed from `service.Producer.Close(ctx)` to `dimensionExtractedProducer.Close(ctx)`

File: main.go [hunk 2]
```diff
@@ -44,18 +52,93 @@ func main() {
...
+		log.Debug("stopped kafka consumer listener", nil)
+		eventLoopCancel()
+		<-eventLoopDone
+		log.Debug("closing http server", nil)
+		if err := api.Close(ctx); err != nil {
+			log.ErrorC("failed to gracefully close http server", err, nil)
+		} else {
+			log.Debug("gracefully closed http server", nil)
+		}
+		log.Debug("closing kafka producer", nil)
```
Comment (nshumoogum) [inline main.go]: We should now be more specific with this producer (dimension extracted producer) as we now have two in the application. This should also be applied to line 126.

File: main.go [hunk 3]
```diff
@@ -44,18 +52,93 @@ func main() {
...
+		service.Producer.Close(ctx)
+		log.Debug("closed kafka producer", nil)
```
Comment (nshumoogum) [inline main.go]: We should be gracefully closing the `dimensionExtractedErrProducer` between lines 126 and 127.

```
log.Debug("closing kafka dimension extracted error producer", nil)
dimensionExtractedErrProducer.Close(ctx)
log.Debug("closed kafka dimension extracted error producer", nil)

File: service/service.go [hunk 4]
```diff
@@ -1,122 +1,208 @@
...
+		line, err := csvReader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.ErrorC("encountered error reading csv", err, log.Data{"instance_id": instanceID, "csv_line": line})
+			return instanceID, err
+		}
+
+		dimension := dimension.Extract{
```
Comment (eldeal) [inline service/service.go]: Should avoid shadowing here as well

File: service/service.go [hunk 5]
```diff
@@ -1,122 +1,208 @@
...
-		eventLoopCancel()
-		<-eventLoopDone
-		log.Debug("Closing http server", nil)
-		if err := api.Close(ctx); err != nil {
-			log.ErrorC("Failed to gracefully close http server", err, nil)
-		} else {
-			log.Debug("Gracefully closed http server", nil)
+	log.Trace("a count of the number of observations", log.Data{"instance_id": instanceID, "number_of_observations": numberOfObservations})
+
+	instance := instance.NewJobInstance(svc.DatasetAPIURL, svc.DatasetAPIAuthToken, instanceID, numberOfObservations, headerRow, svc.MaxRetries)
```
Comment (eldeal) [inline service/service.go]: Shadowing

**Pull request merged**